pkgs/tools/networking/zap/default.nix

```diff
@@ -1,7 +1,25 @@
-{ lib, stdenv, fetchurl, jre, runtimeShell }:
+{ lib, stdenv, fetchurl, jre, runtimeShell, writeScriptBin }:
 
-stdenv.mkDerivation rec {
+let
   pname = "zap";
+  # From https://github.com/zaproxy/zaproxy/blob/master/zap/src/main/java/org/parosproxy/paros/Constant.java
+  version_tag = "2010000";
+  # Copying config and adding version tag before first use to avoid permission
+  # issues if zap tries to copy config on it's own.
+  runner = writeScriptBin "zap" ''
```

Comment:
Did you see `writeShellApplication` it would allow to remove the explicit substitute. It also handle `export PATH=`

Reply:
I am unsure but if you add `JAVA_HOME=jre;` attribute for `writeShellApplication` it will be set as environment variable inside the script.

Reply:
Switched `writeScriptBin` to `writeShellApplication`. 

I don't see any magic that would accept `JAVA_HOME=jre` outside of the script `text`.

As of 
> it would allow to remove the explicit substitute

You mean setting `PATH`? If so - cool :smiley:
If you mean getting rid of `substituteInPlace` - I don't know how to accomplish this to be honest :sweat:

Reply:
> I don't see any magic that would accept JAVA_HOME=jre outside of the script text.

my bad the tricks is only for mkDerivation

> If you mean getting rid of `substituteInPlace`

For removing `substituteInPlace` if you inverse the dependencies you will be able to use string interpolation with `${original_zap}`. Anyway you greatly improved zap :)

Reply:
Ok yes it is a minor change. I already approved the PR but I haven't the right to merge so you can close this remark and wait for a committer to see this PR.